### PR TITLE
Fix issues with .ref in Database functions

### PR DESCRIPTION
**Pull request by samtstern**
### Description

Fixes https://github.com/firebase/firebase-functions/issues/726

TODO:

- [x] Branch on `firebase-functions` version
- [x] merge this PR on `firebase-functions`:  https://github.com/firebase/firebase-functions/pull/727

### Scenarios Tested

```js
const functions = require('firebase-functions');

exports.helloDb = functions.database.ref('/foo').onCreate(async (snap, ctx) => {
  console.log("snap.ref", snap.ref.toString());
  await snap.ref.update({
    'fromFunction': new Date().toISOString()
  });
});
```

### Sample Commands

N/A

### CODE REVIEW

**Comment by samtstern** [comment]
Tests currently failing due to the current validation in `firebase-functions:
```
[default stdout] ⚠  functions: Invalid value for config firebase.databaseURL: http://localhost:9000?ns=fir-dumpster
```

File: src/test/emulators/functionsEmulatorRuntime.spec.ts [hunk 1]
```diff
@@ -391,47 +391,47 @@ describe("FunctionsEmulator-Runtime", () => {
         expect(info.url).to.eql("http://localhost:9090/");
       }).timeout(TIMEOUT_MED);
 
-      it("should merge .initializeApp arguments from user", async () => {
```
Comment (samtstern) [inline src/test/emulators/functionsEmulatorRuntime.spec.ts:394]: Got rid of this wacky test which had been skipped in CI anyway (see lines below) and is no longer useful.

File: src/emulator/functionsEmulatorRuntime.ts [hunk 2]
```diff
@@ -609,10 +609,18 @@ function initializeEnvironmentalVariables(frb: FunctionsRuntimeBundle): void {
     // Ignore, config is optional
   }
 
+  let emulatedDatabaseURL = undefined;
+  if (frb.emulators.database) {
```
Comment (samtstern) [inline src/emulator/functionsEmulatorRuntime.ts:613]: TODO(samstern): This needs to be dependent on a specific version of the `firebase-functions` SDK.  Will have to wait until the functions PR ships to know that version.

File: src/test/emulators/functionsEmulatorRuntime.spec.ts [hunk 3]
```diff
@@ -681,7 +681,7 @@ describe("FunctionsEmulator-Runtime", () => {
 
         const data = await callHTTPSFunction(worker, frb);
         expect(JSON.parse(data)).to.deep.equal({ offset: 0 });
-      });
+      }).timeout(TIMEOUT_MED);
```
Comment (joehan) [inline src/test/emulators/functionsEmulatorRuntime.spec.ts:684]: I might be misunderstanding exactly how mocha's timeout works, but isn't this redundant with the .timeout() on the  next line that covers that whole describe block?

File: src/test/emulators/functionsEmulatorRuntime.spec.ts [hunk 4]
```diff
@@ -391,47 +391,47 @@ describe("FunctionsEmulator-Runtime", () => {
...
-              .onCreate(async (snap: any, ctx: any) => {
-                await admin
-                  .database()
-                  .ref("write-test")
-                  .set({
-                    date: new Date(),
-                  });
-              }),
+            function_id: require("firebase-functions").https.onRequest((req: any, res: any) => {
+              res.json(JSON.parse(process.env.FIREBASE_CONFIG!));
```
Comment (joehan) [inline src/test/emulators/functionsEmulatorRuntime.spec.ts:409]: TIL that you can tell the compiler that something isn't null by adding !

**Comment by joehan** [comment]
Noticed that a the triggers end to end test failed on `  "before all" hook for "should write to the database emulator":`
Might just be a flake, but can you check before you merge?

File: src/test/emulators/functionsEmulatorRuntime.spec.ts (hunk 3 above)
Comment (samtstern) [reply src/test/emulators/functionsEmulatorRuntime.spec.ts]: You'd think, but nope.

**Comment by samtstern** [comment]
@joehan the tests now pass because `firebase-functions` 3.8.0 was released, merging.

**Pull request merged**